[nigelheng] iP

nigelheng opened the pull request:
Replace Duke with Eln in source file
Create echoUser method to repeat user input
Update echoUser method to add user input into list that can be printed

Commit:
Add increment Level-0
Renamed File and Class
Prints Greeting and Farewell

Commit:
Delete Duke.java

Commit:
Add increment Level-1
Add echoUser method
Eln echoes user input till bye is given

Commit:
Add increment Level-2
Update echoUser method
Eln adds user inputs into a list
"list" input prints list of added inputs

sxfoo commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+
+
+public class Eln {
+    static String LINE = "____________________________________________________________";
```
Maybe consider making this private (if it wont be used out of this class) and final (constant)?
```suggestion
    private static final String LINE = 
    "____________________________________________________________";
```

sxfoo replied:
Besides that, I think that you made good note to use capital letters in "LINE" to use for this constant. Nice!

sxfoo commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Eln {
+    static String LINE = "____________________________________________________________";
+    private static void echoUser() {
```
The method name "echoUser" does not match this method's function. Perhaps rename it to something else?
```suggestion
    private static void handleCommands() {
```

sxfoo commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+        while(!input.equals("bye")) {
+            input = scan.nextLine();
+
+            if(input.equals("list")) {
+                System.out.println(LINE);
+                if(index == 0) {
+                    System.out.println("list is empty");
+                }
+                for(int i = 0; i < index; i++) {
+                    System.out.println((i+1) + ". " + list[i]);
+                }
+                System.out.println(LINE);
+            } else {
+                list[index] = input;
+                System.out.println(LINE);
+                System.out.println("added: " + input);
+                System.out.println(LINE);
+                index++;
+            }
+        }
+    }
```
Good note in using K&R style brackets! (aka Egyptian style) for if else statements

sxfoo commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+        String[] list = new String[20];
+        int index = 0;
+
+        while(!input.equals("bye")) {
```
Do remember to include a space between while and the condition!
I noticed this happening for your if and for loop statements down below also
```suggestion
        while (!input.equals("bye")) {
```

jthibault03 replied:
good use of the "happy path" as outlined in the code organization guidelines, since the unique inputs (e.g. bye) are handled in the beginning. would be good to handle error cases around this section too.

sxfoo commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+    private static void echoUser() {
+        Scanner scan = new Scanner(System.in);
+        String input = "";
+        String[] list = new String[20];
```
I like that you kept the array specifiers to the String type and not list!

sxfoo reviewed (commented):
Generally looks good coding standard wise!
Some cosmetic changes requested.

jthibault03 commented on src/main/java/Eln.java:
good how you put LINE in all caps to avoid the "magic string" in your code

jthibault03 commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+                System.out.println(LINE);
+            } else {
+                list[index] = input;
+                System.out.println(LINE);
```
I think it would be good if you created a private method to handle printing different segments, to avoid typing System.out.println(LINE) many times in your code and to create a level of abstraction.

jthibault03 commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,47 @@
+        System.out.println("What can I do for you?");
+        System.out.println(LINE);
+
+        echoUser();
```
I would suggest having more methods than just echoUser() in your Eln class, in order to organize your code a little more

Commit:
Add increment Level-3

implement taskManager
add todo and task classes
add method to mark and unmark tasks

Commit:
Add increment Level-4

add todo, deadline and event classes
add method to print tasks with task type

Commit:
Add increment Level-5

add ElnException abstract class
add errors for invalid todo, deadline and event.
add help message

Commit:
Merge branch 'branch-Level-5'

Musfirahe0556596 commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,105 @@
+
+public class Eln {
+    private static final String LINE = "____________________________________________________________";
+    private static Task[] tasksList = new Task[100]; // max number of tasks set to 100
```
Plural form should be used on names representing a collection of objects! Also, a suggestion: can put the number 100 as a named constant :]

Musfirahe0556596 commented on src/main/java/Eln.java:
```diff
@@ -0,0 +1,105 @@
+            newTask = new Deadline(taskToAdd);
+        } else if (taskToAdd.startsWith("event")) {
+            newTask = new Event(taskToAdd);
+        } else {
+            newTask = new Task(taskToAdd);
+        }
```
might want to be careful with putting this as the default option/else loop. A task can either be a todo, deadline or event - but if user inputs some other string, a (general) Task is added which does not fall in any of the task types mentioned!

Musfirahe0556596 reviewed (commented):
Code seems oki (do implement your increments!)! Just some comments to take note of!

Commit:
Create .jar file for tutorial

Commit:
Add increment Level-6

Implement delete and moreOOP

Commit:
Add increment Level-7

Implement ability to save

Commit:
Merge branch 'branch-Level-6'

Commit:
Solve Merge Conflict

Commit:
Add Increment Level-9

Implement Find to search list for keywords

Commit:
Revert "Add Increment Level-9"

This reverts commit da7b92808f586c09711e196e710c0d8579710144.

Commit:
Add Increment Level-9

Implement Find to search list for keywords

Commit:
Add Increment A-JavaDoc

Add JavaDoc comments

nigelheng committed:
Merge pull request #1 from nigelheng/branch-Level-9

Add Increment Level-9

nigelheng committed:
Merge branch 'master' into branch-A-JavaDoc

nigelheng committed:
Merge pull request #2 from nigelheng/branch-A-JavaDoc

Add Increment A-JavaDoc

Commit:
Merge branch 'master' of https://github.com/nigelheng/ip

Commit:
Add increment A-UserGuide

Implemented userguide in README

Commit:
Add print lines when adding task
